### PR TITLE
Add CED model for audio classification

### PR DESCRIPTION
**Pull request by jimbozhang**
# What does this PR do?

Add a new model for audio classification: CED. 

CED are simple ViT-Transformer-based models, which were proposed in [CED: Consistent ensemble distillation for audio tagging](https://arxiv.org/abs/2308.11957). Notable differences from other available models (such as [AST](https://huggingface.co/docs/transformers/model_doc/audio-spectrogram-transformer)) include:
1. Simplification for finetuning: Batchnormalization of Mel-Spectrograms. During finetuning one does not need to first compute mean/variance over the dataset, which is common for AST.
1. Support for variable length inputs. Most other models use a static time-frequency position embedding, which hinders the model's generalization to segments shorter than 10s. Many previous transformers simply pad their input to 10s in order to avoid the performance impact, which in turn slows down training/inference drastically.
1. Training/Inference speedup: 64-dimensional mel-filterbanks and 16x16 patches without overlap, leading to 248 patches from a 10s spectrogram. In comparison, AST uses 128 mel-filterbanks with 16x16 (10x10 overlap) convolution, leading to 1212 patches during training/inference. CED-Tiny runs on a common CPU as fast as a comparable MobileNetV3.
1. Performance: CED-Mini with 10M parameters outperforms the majority of previous approaches (~80M).

Here is a [notebook](https://colab.research.google.com/drive/1MOdY7t3qSN6ZTDtYqrkp991nYH6KqLS-) illustrates how to use CED to detect audio events.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Models:

- speech models: @sanchit-gandhi

### CODE REVIEW

**Comment by ArthurZucker** [comment]
fyi @sanchit-gandhi

**Comment by jimbozhang** [comment]
Hi there~ 👋

I submitted this pull request a little while ago, and I was wondering if someone could take a look at it. I've already tagged some potential reviewers, but I haven't received any feedback yet. I'm looking forward to hearing from you whenever you have a chance. 😊

BTW, I'm not sure why `ExamplesTestsNoTrainer.test_run_image_classification_no_trainer` [failed in CircleCI test](https://app.circleci.com/pipelines/github/huggingface/transformers/77686/workflows/9d6c662b-2d36-433a-9f92-2f189388b7dc/jobs/989529/parallel-runs/0/steps/0-116).

**Review by ArthurZucker — COMMENTED** [review]
Hey! Thanks for the PR! 🤗 
CED seems to be a training framework, which in this case would not really have it's place in the transformers library. If what you are adding is a model that was trained using this framework but that is not already available in the library, then the name should be that of the model. 

ViT's  are however already implemented, and a few audio models are also based on them! (see CLAP for example).

My recommendation for now would be to put the models on the hub following [this](https://huggingface.co/docs/transformers/custom_models) tutorial
